pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -154,32 +156,36 @@ func (c *Cluster) checkHealth() error {
 				delete(c.monTimeoutList, mon.Name)
 				logger.Infof("mon %s is back in quorum, removed from mon out timeout list", mon.Name)
 			}
-		} else {
-			logger.Debugf("mon %s NOT found in quorum. Mon quorum status: %+v", mon.Name, quorumStatus)
-			allMonsInQuorum = false
+			continue
+		}
 
-			// If not yet set, add the current time, for the timeout
-			// calculation, to the list
-			if _, ok := c.monTimeoutList[mon.Name]; !ok {
-				c.monTimeoutList[mon.Name] = time.Now()
-			}
+		logger.Debugf("mon %s NOT found in quorum. Mon quorum status: %+v", mon.Name, quorumStatus)
+		allMonsInQuorum = false
 
-			// when the timeout for the mon has been reached, continue to the
-			// normal failover/delete mon pod part of the code
-			if time.Since(c.monTimeoutList[mon.Name]) <= MonOutTimeout {
-				timeToFailover := int(MonOutTimeout.Seconds() - time.Since(c.monTimeoutList[mon.Name]).Seconds())
-				logger.Warningf("mon %s not found in quorum, waiting for timeout (%d seconds left) before failover", mon.Name, timeToFailover)
-				continue
-			}
+		// If not yet set, add the current time, for the timeout
+		// calculation, to the list
+		if _, ok := c.monTimeoutList[mon.Name]; !ok {
+			c.monTimeoutList[mon.Name] = time.Now()
+		}
+
+		// when the timeout for the mon has been reached, continue to the
+		// normal failover/delete mon pod part of the code
+		if time.Since(c.monTimeoutList[mon.Name]) <= MonOutTimeout {
+			timeToFailover := int(MonOutTimeout.Seconds() - time.Since(c.monTimeoutList[mon.Name]).Seconds())
+			logger.Warningf("mon %s not found in quorum, waiting for timeout (%d seconds left) before failover", mon.Name, timeToFailover)
 
-			logger.Warningf("mon %s NOT found in quorum and timeout exceeded, mon will be failed over", mon.Name)
-			c.failMon(len(quorumStatus.MonMap.Mons), desiredMonCount, mon.Name)
-			// only deal with one unhealthy mon per health check
-			return nil
+			// Restart the mon if it is stuck on a failed node
+			c.restartMonIfStuckTerminating(mon.Name)
```

Comment:
Just to be sure, at this point, the mon is out of quorum and we just force delete it, correct?

Reply:
Yes that's correct